src/main/java/org/passay/PasswordValidator.java

```diff
@@ -111,6 +111,7 @@ public RuleResult validate(final PasswordData passwordData)
         result.setValid(false);
         result.getDetails().addAll(rr.getDetails());
       }
+      result.getMetadata().putAll(rr.getMetadata().getAll());
```

Comment:
Do you want to accommodate value merging here? I don't believe we have a case at present where two different rules produce the same key, but in that case the implementation above would cause the latter to overwrite the former. Merge semantics would be preferable. On the other hand we can say "don't do that."

Reply:
I prefer don't do that. Rules participating in metadata should produce unique keys.

Reply:
That's fine.